### ISSUE TITLE
Simplify `add` and `link` interfaces

### ISSUE THREAD

**Issue by xaviervia** [xaviervia/sydney#24]
Derived from #23 . Only two interfaces will remain now:

### 1. Another `Sydney`
```javascript
var base = new Sydney
var another = new Sydney

base.link(another)
```

### 2. Vanilla, to amplify
```javascript
var base = new Sydney
var another = {
  callback: function () {}
}

base.link(another)
```

... in which case `another` is amplified with all the `Sydney` methods. If it had a property of the `Sydney` interface already in place, it is not overriden:

```javascript
var base = new Sydney
var another = {
  send: "hello"
}

base.link(another)

another.send // => "hello"
```

And that's it. No other extension to these two, not even support for not extending the module with the mixin methods.

With this in mind, it might be good to keep Sydney's interface minimal. Right now it includes:

- `send`
- `broadcast`
- `link`
- `unlink`
- `add`
- `remove`

**Comment by xaviervia** [comment]
By extension this will simplify `unlink` and remove` interfaces, since they only have to match the subscriber, with no need for searching for it by comparing `callback` and `endpoint`. The "static" method `Sydney.find` might be rendered obsolete.

The constructor interface should also be updated: it should only support `(endpoint, callback)`, `(endpoint)` and `(callback)`. `Sydney.make` might be still useful, or not.

**Comment by xaviervia** [comment]
What about the `subscribers` property? Should it be overriden? Should it be extra type-checked when used?

This _mixin-safe_ thing is quite complicated.

I think it should be left "as is" if it exists. Add a test for that.

**Comment by xaviervia** [comment]
Solved in `v1.0.0`!
